src/deriver/api.py

```diff
@@ -357,6 +362,90 @@ def scan_selfies(self):
 
         return good_children, all_filtered_children
 
+    def derive_smiles_gb(self, n_children: int = 100, mut_rate: float = 0.01):
+        children = []
+        good_children = []
+
+        if self.data.filter:
+            filter_params = self.data.filter_params
+        else:
+            logger.warning("Warning: No filter has been set, so all child molecules will be labeled"
+                           " as 'good' regardless of quality. Please call Deriver.set_filter() first"
+                           " in order to use a filter for drug-likeness.")
+            filter_params = None
+
+        for _ in range(n_children):
+            parent_a, parent_b = np.random.choice(self.data.seed_smiles)
+            try:
+                new_child = co.crossover(parent_a, parent_b)
+                if new_child is not None:
+                    new_child = mu.mutate(new_child, mut_rate)
+                assert new_child
+                children.append(new_child)
+            except:
+                continue
+        child_mols = [Chem.MolFromSmiles(child, sanitize=True) for child in children]
+        filtered_children = apply_filter(filter_params,
+                                         child_mols,
+                                         self.data.must_have_patterns,
+                                         self.data.must_not_have_patterns
+                                         )
+        for child in filtered_children:
+            if filtered_children[child]["is_good"]:
+                # check the cache
+                if self.data.filter_molecules:
+                    if child not in self.data.filter_molecules:
+                        good_children.append(child)
+                    else:
+                        logger.debug(f"skipping previously seen molecule: {child}")
+                else:
+                    good_children.append(child)
+
+        logger.info(f"Generated {len(good_children)} 'good' children.")
+        return good_children, filtered_children
```

Comment:
needs to also add to a Deriver.Data field for molecules from this method, and to clear it at the beginning